### ISSUE TITLE
Minus Max of Distance

### ISSUE THREAD

**Issue by pakheiyeung** [CYang0515/pytorch_ssn#8]
Hi, at line 212 of util.py, why pixel_spixel_neg_dist needs to minus the maximum of pixel_spixel_neg_dist along dimension 1? In the paper, I think they directly do exponential of the negative distance to compute Q. Thanks

**Comment by CYang0515** [comment]
It's for the numerical stability. Because the pixel_spixel_neg_dist may be a large value. In original code, they use the softmax function in Caffe and it minus the maximum value.

**Comment by pakheiyeung** [comment]
Got it. Thank you very much!
